extensions/users-permissions/models/User.js

```diff
@@ -5,9 +5,24 @@ module.exports = {
    * Triggered before user creation.
    */
   lifecycles: {
+    // async beforeUpdate(params, data) {
+    //   if (data.username)
+    //     data.username = data.username.trim().replace(/\s+/g, '_');
+    // },
     async beforeUpdate(params, data) {
-      if (data.username)
+      const {username, firstName, lastName} = data;
+      if (username !== null && username !== undefined) {
```

Comment:
You can simplify this check to https://stackoverflow.com/questions/5515310/is-there-a-standard-function-to-check-for-null-undefined-or-blank-variables-in.